app/src/main/java/com/edulectronics/tinycircuit/ui/CircuitActivity.java

```diff
@@ -66,20 +77,66 @@ public boolean onTouch(View v, MotionEvent event) {
 
     /*This code adds a menu to the side*/
     private void makeMenu(){
-        ListView menu = (ListView) findViewById(R.id.menu);
-        String[] items = getResources().getStringArray(R.array.items);
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(
-                this, android.R.layout.simple_list_item_1, items);
-        menu.setAdapter(adapter);
-
-        menu.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-            public void onItemClick(AdapterView<?> adapterView, View view,
-                                    int position, long itemId){
-                /*Not implemented, should select desired item*/
+        ExpandableListView expandableList = (ExpandableListView) findViewById(R.id.expandablelist);
+        makeLists();
+
+        ExpandableListAdapter adapter = new ExpandableListAdapter(
+                this, headers, children, expandableList);
+        expandableList.setAdapter(adapter);
+
+        expandableList.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
```

Comment:
Liever de on click handlers in een aparte methode zetten. Iets als expandableList.setOnChildClickListener(this.iets(*paramaters))